.github/workflows/wheel_tests_and_release.yml

```diff
@@ -48,6 +45,26 @@ jobs:
           python -m build --no-isolation --skip-dependency-check --sdist .
           ls -la ${{ github.workspace }}/dist
 
+      - name: Generate artifact attestation for sdist and wheels
+        uses: actions/attest-build-provenance@173725a1209d09b31f9d30a3890cf2757ebbff0d # v1.1.2
```

Comment:
Would it make sense to create the attestation as close to artifact generation as possible, e.g. in the job `call-workflow-build-wheels`? Ideally, before anything else like the test suite is run on the created wheel?

Reply:
@lagru Good question/point. I think that to some degree I would say yes, but if you want to only create attestations for releases and not just every run of the CI then you would either need to totally refactor the

```yaml
jobs:
  call-workflow-build-wheels:
    uses: ./.github/workflows/wheels_recipe.yml
```

workflow to know about release situations (probably not worth it) or just do the slightly less perfect situation of this PR (signing the wheels just after download...the sdist is signed immediatley after build here at least).

Reply:
Okay, I thing we can probably come up with a solution that uses `inputs` to enable artifacts in `wheels_recipe` only for releases. But let's do that in another PR (I'll open an issue) and merge this for now.